src/router/chainable_router.rs

```diff
@@ -0,0 +1,71 @@
+use hyper::method::Method;
+use middleware::Middleware;
+use nickel::Nickel;
+use router::http_router::{HttpRouter};
+
+pub struct ChainableRoute<'a, D: Sync + Send + 'static = ()> {
+    path: String,
+    nickel: &'a mut Nickel<D>
+}
+
+pub trait ChainablePath<'a, D> {
+    fn route(&'a mut self, path: &str) -> ChainableRoute<'a, D>;
+}
+
+impl<'a, D: Sync + Send + 'static> ChainablePath<'a, D> for Nickel<D> {
+    fn route(&'a mut self, path: &str) -> ChainableRoute<'a, D> {
+        ChainableRoute {
+            path: path.to_string(),
+            nickel: self
+        }
+    }
+}
+
+pub trait ChainableHandler<'a, D> {
```

Comment:
I think these methods could just be inherent methods, as it seems a bit much to require two trait imports for one feature?